pkg/services/playlist/playlistimpl/sqlx_store.go

```diff
@@ -0,0 +1,248 @@
+package playlistimpl
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/playlist"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/jmoiron/sqlx"
+)
+
+type sqlxStore struct {
+	sqlxdb  *sqlx.DB
+	dialect migrator.Dialect
+}
+
+func (s *sqlxStore) WithTransaction(ctx context.Context, db *sqlx.DB, fn func(*sqlx.Tx) error) error {
+	// In the same transaction is quaranteed in the same session
+	tx, err := db.Beginx()
+	if err != nil {
+		return err
+	}
+	// Here when I try to set the session sql_mode, other integration test is failed
+	if s.dialect.DriverName() == "mysql" {
+		_, err = tx.ExecContext(ctx, "SET SESSION sql_mode = 'ANSI_QUOTES'")
+		if err != nil {
+			return err
+		}
+	}
+	err = fn(tx)
+	if err != nil {
+		if rbErr := tx.Rollback(); rbErr != nil {
+			return fmt.Errorf("tx err: %v, rb err: %v", err, rbErr)
+		}
+		return err
+	}
+	return tx.Commit()
+}
+
+func (s *sqlxStore) insertWithReturningId(ctx context.Context, tx *sqlx.Tx, p *playlist.Playlist) error {
```

Comment:
we do need this returning id, since playlist_items table is linked to playlist table by id, and we can't do get here, since it is in a transaction, it is not committed yet.

Reply:
this "insert query builder" should go to dialect probably